WEBPACK.md

```diff
@@ -0,0 +1,144 @@
+# Webpack configuration file
+
+This file provides a documentation for `webpack.config.js`
+
+```javascript
+// Include necessary modules
+const webpack = require('webpack');
+const glob = require('glob');
+const path = require('path');
+const MergeIntoSingleFilePlugin = require('webpack-merge-and-include-globally');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+
+const config = {
+  // "production" or "development", set by cross-env in NPM scripts
+  // Determines which optimizations it needs to apply to our JavaScript files
+  mode: process.env.NODE_ENV,
+  // Files to bundle(in order from the top to bottom)
+  entry: [
+    /** glob.sync(folder) returns an array of file paths in the folder
+     * ...(spread) operator spreads out an array
+     * ["f1", "f2"] -> "f1", "f2"
+     * entry only accepts an array of strings(files) to bundle or a single entry file
+     */
+    ...glob.sync('./src/util/*'),
+    './src/DistortableImageOverlay.js',
+    './src/DistortableCollection.js',
+    './src/edit/getEXIFdata.js',
+    './src/edit/handles/EditHandle.js',
+    ...glob.sync('./src/edit/handles/*', {
+      ignore: './src/edit/handles/EditHandle.js'
+    }),
+    './src/iconsets/IconSet.js',
+    './src/iconsets/KeymapperIconSet.js',
+    './src/iconsets/ToolbarIconSet.js',
+    './src/edit/actions/EditAction.js',
+    ...glob.sync('./src/edit/actions/*', {
+      ignore: './src/edit/actions/EditAction.js'
+    }),
+    './src/edit/toolbars/DistortableImage.PopupBar.js',
+    './src/edit/toolbars/DistortableImage.ControlBar.js',
+    './src/edit/DistortableImage.Edit.js',
+    './src/edit/DistortableCollection.Edit.js',
+    './src/components/DistortableImage.Keymapper.js',
+    './src/mapmixins/DoubleClickZoom.js',
+    ...glob.sync('./src/mapmixins/*', {
+      ignore: './src/mapmixins/DoubleClickZoom.js'
+    })
+  ],
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: 'leaflet.distortableimage.js',
+    /**
+     * Webpack live server, on change in the source code, compiles files again and 	     * generates new hot update files. Instead of generating new files on each change,  * we just override the previous file.
```

Comment:
I was wondering if you could explain this one to me please? I tested it without those lines and saw what you meant, but what are the implications of the change?  On webpack docs I see it says typically a project won't need to use these attributes

Reply:
Thank you for the interest, I'll explain, of course!

You've tried it and you saw how it generates many of xyz.json files. That's good, now let me explain why that happens.
This is due to HMR (Hot Module Replacement), `hot: true` option.

HMR is a Webpack feature that allows us to update our modules/files without a browser reload! It mostly shines on bigger projects where loaders are used, but I think we can benefit from it also.
For example, we can update our console.log when we are debugging without losing our apps state(browser reloads and everything is gone).

It's just an optional feature and I can remove it if you want me to do so.

Reply:
were good to leave this! I was just trying to figure out, what are the implications of overriding the previous file as opposed to letting it compile multiple files?

Reply:
It results in high I/O and it bloats our dist folder.
Working without this option would create 100+ these files in `dist/hot` folder in less than 10 minutes of coding. This slows down your IDE because it needs to update our working directory every time we change something.

That's why I decided to put these options.

Reply:
That's what I figured! You would think everyone would want to override this option then?

Reply:
I think that it's a good option to have.
If we have great machines, it doesn't mean that everyone has them.

I can bet that this would be a huge problem to an average laptop/computer with 4GB RAMs, especially if you are using Visual Studio or Visual Studio Code + Chrome/Firefox and you are fried :)

Reply:
gotchya, thank you!! Resolving!